source/common/network/cidr_range.cc

```diff
@@ -78,24 +78,53 @@ IpVersion CidrRange::version() const {
   return IpVersion::v4;
 }
 
-bool CidrRange::isInRange(InstanceConstSharedPtr address) const {
-  if (address == nullptr) {
+bool CidrRange::isInRange(const Instance& address) const {
+  if (address_ == nullptr || !isValid() || address.type() != Type::Ip ||
+      version() != address.ip()->version()) {
     return false;
   }
 
-  return isInRange(*address);
-}
+  // All addresses in range.
+  if (length_ == 0) {
+    return true;
+  }
 
-bool CidrRange::isInRange(const Instance& address) const {
-  if (length_ < 0 || address.type() != Type::Ip || address.ip()->version() != version()) {
-    return false;
+  switch (address.ip()->version()) {
+  case IpVersion::v4:
+    if (ntohl(address.ip()->ipv4()->address()) >> (32 - length_) ==
+        ntohl(ipv4()->address()) >> (32 - length_)) {
+      return true;
+    }
+    break;
+  case IpVersion::v6:
+    int length = length_;
+    // Loop through address bytes and compare. Address is in network byte order.
+    for (int i = 0; i < 16; i++) {
+      if (length == 0) {
+        return true;
+      } else if (length < 8) {
+        // Compare relevant bits.
+        if (address.ip()->ipv6()->address()[i] >> (8 - length) ==
+            ipv6()->address()[i] >> (8 - length)) {
+          return true;
+        } else {
+          break;
```

Comment:
It's probably clearer to replace `break` with `return false` here (and in this case, just replace the `if` statement with a single `return`).